cmake/onnxruntime_providers.cmake

```diff
@@ -304,6 +304,9 @@ if (onnxruntime_USE_TENSORRT OR onnxruntime_USE_DNNL)
   if(APPLE)
     set_property(TARGET onnxruntime_providers_shared APPEND_STRING PROPERTY LINK_FLAGS "-Xlinker -exported_symbols_list ${ONNXRUNTIME_ROOT}/core/providers/shared/exported_symbols.lst")
   elseif(UNIX)
+    if (onnxruntime_USE_TENSORRT)
+      set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -Wno-deprecated-declarations")
```

Comment:
Who needs this? Which module?

Reply:
TRT reports error,
TensorRT-7.1.3.4/include/NvInfer.h:5565:81: error: ‘IPluginLayer’ is deprecated [-Werror=deprecated-declarations]
         ITensor* const* inputs, int nbInputs, IPluginExt& plugin) TRTNOEXCEPT = 0;